stripped/colabs/Regresja_logistyczna_i_SVM_stripped.py

```diff
@@ -0,0 +1,1180 @@
+August_2019_DF.reset_index(inplace=True, drop=True)
+
+August_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==8])
+August_2021_DF.reset_index(inplace=True, drop=True)
+
+December_2019_DF = pd.DataFrame(data = nasa_2019_an[nasa_2019_an['Month']==12])
+December_2019_DF.reset_index(inplace=True, drop=True)
+
+December_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==12])
+December_2021_DF.reset_index(inplace=True, drop=True)
+
+#Funkcja do generowania podziału na foldy do crosswalidacji blokowej (spatial block crossvalidation)
+#kod ze sprintu 1 musiałam zaadaptować pod Pandasa
+
+
+def get_grid(grid_cell_size: float, min_lat: float, max_lat: float,
+             min_lon: float, max_lon: float) -> Tuple[List[float], List[float]] :
+ ''' grid_cell_size - the grid cell size (the length of the side of
+                         a square) in degrees '''
+
+ area = (max_lat-min_lat)*(max_lon-min_lon)
+ cells_num = area//(grid_cell_size*grid_cell_size)
+ actual_grid_size = math.sqrt(area/cells_num)
+
+ xx=np.arange(min_lon, max_lon, step = actual_grid_size)
+ yy=np.arange(min_lat, max_lat, step = actual_grid_size)
+
+ return xx, yy
+
+def block_partition(df: pd.DataFrame, block_size: float, min_lat: float,
+                    max_lat: float, min_lon: float, max_lon: float) -> pd.DataFrame :
+  ''' block_size: approximate size of a block '''
+
+  blocked_df = copy.deepcopy(df)
+  block_number = np.zeros(len(df))
+  xx, yy = get_grid(block_size, min_lat, max_lat, min_lon, max_lon)
+
+  block_count = 0
+  for y in range(len(yy)-1):
+    for x in range(len(xx)-1):
+      block = df[(xx[x]<=df['lon']) & (df['lon'] <= xx[x+1]) & ( yy[y]<= df['lat']) & (df['lat'] <= yy[y+1])]
+      if len(block)>0:    #checking if block is non-empty
+         block_number[list(block.index)] = block_count
+         block_count+=1
+
+  blocked_df['block'] = block_number
+  return blocked_df
+
+
+def Kfolds(df: pd.DataFrame, k : int, random_state: Optional[int]) -> List[pd.DataFrame]:
+  ''' k - number of folds '''
+
+  blocks_num = int(df['block'].max())
+  blocks_list = [i for i in range(blocks_num+1)]
+  n = (blocks_num+1)//k
+  reminder = len(blocks)%k
+  folds=[]
+
+  for i in range(k):
+      fold = []
+      for j in range(n):
+          if random_state:
+             random.seed(random_state)
+          r = random.randint(0, len(blocks_list)-1)
+          fold.append(blocks_list[r])
+          blocks_list.remove(blocks_list[r])
+      folds.append(fold)
+
+  if reminder!=0:
+     for b in blocks_list:
+        if random_state:
+           random.seed(random_state)
+        n_fol = random.randint(0, k-1)
+        folds[n_fol].append(b)
+        blocks_list.remove(b)
+
+  foldsDF_list = []
+  folds_data = [df[df['block'].isin(fold)] for fold in folds]
+  foldsDF_list = [pd.DataFrame(data=fold_data) for fold_data in folds_data]
+  foldsDF_list = [foldDF.drop(columns=['block']) for foldDF in foldsDF_list]
```

Comment:
Po co mnożyć linijki i kod?
```suggestion
  foldsDF_list = [pd.DataFrame(data=fold_data).drop(columns=['block']) for fold_data in folds_data]
```

Reply:
Myślałam, że tak będzie czytelniej, bo kiedys dostałam uwagę o małej czytelności kodu, który robił kilka rzeczy w jednej linijce... Tylko dlatego ;)

Reply:
Tym razem za mocno w drugą stronę.

Reply:
Ok.